OpenStack-Rabbit-Consumer/tests/test_openstack_api.py

```diff
@@ -125,7 +123,21 @@ def test_get_server_networks(address, server_details, vm_data):
 
 
 @patch("rabbit_consumer.openstack_api.get_server_details")
-def test_get_server_networks_no_internal(server_details, vm_data):
+@patch("rabbit_consumer.openstack_api.OpenstackAddress")
+def test_get_server_networks_services(address, server_details, vm_data):
+    """
+    Test that the function calls the correct functions to get the networks of a VM
+    """
+    server_details.return_value.addresses = {"Services": []}
+
+    get_server_networks(vm_data)
+    address.get_services_networks.assert_called_once_with(
+        server_details.return_value.addresses
+    )
+
+
+@patch("rabbit_consumer.openstack_api.get_server_details")
+def test_get_server_networks_no_network(server_details, vm_data):
     """
     Tests that an empty list is returned when there are no internal networks
     """
```

Comment:
Does this test also cover the case when there are no internal and no services networks?

Reply:
Yes it does cover that, I've removed the public network placeholder from the dictionary so it is more clear